.github/workflows/ci.yml

```diff
@@ -50,3 +52,20 @@ jobs:
         cd bench
         cabal bench -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter"
 
+  build-freebsd:
+    runs-on: macos-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Build
+      id: build-freebsd
+      uses: vmactions/freebsd-vm@v0.0.8
+      with:
+        usesh: true
+        prepare: pkg install -y ghc hs-cabal-install
+        run: |
+          freebsd-version
+          cabal update
+          cabal build
+          (cd tests; cabal test)
+          (cd bench; cabal bench -O0 --benchmark-options "--quick --min-duration=0 --include-first-iter")
```

Comment:
If there isn't a particular reason to exclude the `sdist` and `haddock` checks, I'd suggest to include them.

Reply:
No real reason, I just copy-pasted this command from another place. Let me add sdist and haddock.